transport/response.hh

```diff
@@ -66,6 +66,21 @@ public:
         }
     }
 
+    response(int16_t stream, cql_binary_opcode opcode, const tracing::trace_state_ptr& tr_state_ptr,
+             const std::map<sstring, bytes>& custom_payload)
+            : _stream{stream}
+            , _opcode{opcode}
+    {
+        if (tracing::should_return_id_in_response(tr_state_ptr)) {
+            auto i = _body.write_place_holder(utils::UUID::serialized_size());
+            tr_state_ptr->session_id().serialize(i);
+            set_frame_flag(cql_frame_flags::tracing);
+        }
```

Comment:
Calling the other constructor would avoid having duplicate `should_return_id_in_response...` code in here:

```cpp
response(int16_t stream, cql_binary_opcode opcode, const tracing::trace_state_ptr& tr_state_ptr,
             const std::map<sstring, bytes>& custom_payload) 
        : response(stream, opcode, tr_state_ptr) 
{
        write_bytes_map(custom_payload);
        set_frame_flag(cql_frame_flags::custom_payload);
}
```

Apart from this, LGTM and we can merge.